auth/src/main/java/com/firebase/ui/auth/viewmodel/idp/LinkingProviderResponseHandler.java

```diff
@@ -33,11 +33,33 @@ protected void signIn(@NonNull AuthCredential credential,
         FirebaseUser currentUser = getCurrentUser();
         if (currentUser == null) {
             getAuth().signInWithCredential(credential)
-                    .addOnSuccessListener(new StartLink(response))
-                    .addOnFailureListener(new OnFailureListener() {
+                    .continueWithTask(new Continuation<AuthResult, Task<Void>>() {
                         @Override
-                        public void onFailure(@NonNull Exception e) {
-                            setResult(Resource.<IdpResponse>forFailure(e));
+                        public Task<Void> then(@NonNull Task<AuthResult> task) {
+                            if (mRequestedSignInCredential == null) {
+                                return Tasks.forResult(null);
```

Comment:
Can you add a comment here explaining why in this case we don't care at all if the `signInWithCredential` operation was successful or not?

Reply:
Nope, that was a mistake, thanks for catching it!